client/src/reducers/teaReducers.js

```diff
@@ -1,20 +1,23 @@
-import { Set, Map } from "immutable";
-
-export default (state = [], action) => {
+export default (state = { allTeas: {}, teaIDs: [] }, action) => {
   switch (action.type) {
     case "ADD_TEA":
-      return [...state, action.payload];
-    case "DELETE_TEA":
-      return state.filter(t => t.id !== action.payload.id);
     case "EDIT_TEA":
-      return state.map(t => (t.id === action.payload.id ? action.payload : t));
+      let updateAllTeas = {
+        ...state.allTeas,
+        ...{ [action.payload.id]: action.payload }
```

Comment:
This line can be simplified to `[action.payload.id]: action.payload` (i.e. remove the spread operator)